.gitignore

```diff
@@ -1,4 +1,42 @@
+### Bazel ###
+# gitignore template for Bazel build system
+# website: https://bazel.build/
+
+# Ignore all bazel-* symlinks. There is no full list since this can change
+# based on the name of the directory bazel is cloned into.
 /bazel-*
 
-# Rust artifacts.
-/**/target
+### Rust ###
+# Generated by Cargo
+# will have compiled files and executables
+/**/target/
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
+# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
+Cargo.lock
```

Comment:
We should not ignore Cargo.lock, it contains the list of specific dependencies used to reproducibly build our examples. I think the place you got this .gitignore from assumes that you are developing a Rust library, in which case you would typically not check in Cargo.lock, just Cargo.toml

Reply:
OK, I removed that part.